mqtt_locator.py

```diff
@@ -37,11 +39,15 @@ def life_factor(self):
 
     def draw(self, win):
         """Renders a ping to a display window"""
-        radius = int(round(self.life_factor() * self.size))
+        radius = int(round((1-self.life_factor()) * self.size))
         thickness = 2
         if thickness > radius:
             thickness = radius
-        pygame.draw.circle(win, self.color, self.coordinate, radius, thickness)
+        pygame.draw.rect(win, self.color, (self.coordinate[0] - radius/2, self.coordinate[1] - radius/2, radius, radius), 0)
```

Comment:
I think it's less obvious what is happening here, I'd love to see the logic broken out into more digestible chunks!